app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,20 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+
+    respond_to do |format|
+      if @scenario.agents.update_all disabled: (params[:scenario][:disabled] == 'true' ? true : false)
```

Comment:
I think `update_all` returns true when success. Mysql and pg handled boolean differently. From what I can tell, the statement is to test whether `params[:scenario][:disabled]` has `true` for value and if yes then ruby `true` otherwise then is ruby `false`. Its written this way to solve the pg/mysql bug and not necessarily to save `update_all` if it fails. So is this still wrong? @dsander had a solution to use  a hidden checkbox whose value rails automatically casts to a boolean. I can try that instead.